csharp/src/Drivers/Apache/Hive2/HiveServer2Statement.cs

```diff
@@ -426,12 +427,128 @@ private async Task<QueryResult> GetQueryResult(TSparkDirectResults? directResult
                 int columnCount = HiveServer2Reader.GetColumnCount(rowSet);
                 int rowCount = HiveServer2Reader.GetRowCount(rowSet, columnCount);
                 IReadOnlyList<IArrowArray> data = HiveServer2Reader.GetArrowArrayData(rowSet, columnCount, schema, Connection.DataTypeConversion);
+
+                // Enhance column schema results if this is a GetColumns query
+                if (SqlQuery?.ToLowerInvariant() == GetColumnsCommandName)
```

Comment:
I think you can refactor this method so that you don't need to test if the current query is the `GetColumnsCommandName`. The `EnhanceGetColumnsResult` should be called from within `GetColumnsAsync`.

I can propose a refactor that uses virtual implementations, if you want a more concreate suggestion.